### ISSUE TITLE
Missing headers on request in production environment

### ISSUE THREAD

**Issue by dodedodo33** [phpro/soap-client#256]
I am developing a small wordpress plugin to forward orders to a supplier.
For this I have setup a local development environment and everything works fine.
Now when testing on the production environment I get a long delay and then I get a TypeError on a MixedType response not matching the specific response for the called action. Using the logging event handler with monolog I discovered that the client recorded an empty response, hence why the MixedType response was returned.

I have tried to track down the issue by comparing both environments and trying out different versions of php and what not. When using a configuration that is exactly the same in terms of php version and the ext-soap and ext-xml enabled and both same versions the problem still exists.

I've tried debugging further by comparing all inputs to the client and the client configuration upon sending the request and they all seemed to match. Also the wsdl file is loaded with the local wsdl provider using the correct path and the exact same wsdl. The wsdl does have two bindings, one for SOAP 1.1 and one for SOAP 1.2. Both resolve to the same url and pinning the SOAP version used by the client made no difference.

The last thing I tried was enabling the trace option upon client creation. Then checking the debug info from $client->debugLastSoapRequest();

That's where things got interesting, on my local installation I got the following:
``` php
Array
(
    [request] => Array
        (
            [headers] => POST [REDACTED] HTTP/1.1
Host: [REDACTED]
Connection: Keep-Alive
User-Agent: PHP-SOAP/7.3.8
Content-Type: application/soap+xml; charset=utf-8; action="[REDACTED]"
Content-Length: 1186
            [body] => <?xml version="1.0" encoding="UTF-8"?>
<env:Envelope xmlns:env="http://www.w3.org/2003/05/soap-envelope" xmlns:ns1="[REDACTED]">
  <env:Body>
    [REDACTED]
  </env:Body>
</env:Envelope>

        )

    [response] => Array
        (
            [headers] => HTTP/1.1 200 OK
Cache-Control: private, max-age=0
Content-Type: application/soap+xml; charset=utf-8
Server: Microsoft-IIS/8.5
X-AspNet-Version: 4.0.30319
X-Powered-By: ASP.NET
Date: Mon, 26 Aug 2019 12:05:25 GMT
Content-Length: 644
            [body] => <?xml version="1.0" encoding="utf-8"?>
<soap:Envelope xmlns:soap="http://www.w3.org/2003/05/soap-envelope" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
  <soap:Body>
    [REDACTED]
  </soap:Body>
</soap:Envelope>

        )

)
```

On the production environment I get this:
``` php
(
[request] => Array
(
[headers] =>
[body] => <?xml version="1.0" encoding="UTF-8"?>
<env:Envelope xmlns:env="http://www.w3.org/2003/05/soap-envelope" xmlns:ns1="[REDACTED]">
    <env:Body>
        [REDACTED]
    </env:Body>
</env:Envelope>

)

[response] => Array
(
[headers] =>
[body] =>
)

)
```

So it seems the request is missing all headers.

### Support Question

Does anyone maybe have a suggestion what the problem could be or how I can debug further?
I can't find any real differences between the two environments.

**Comment by veewee** [comment]
The empty response is due to an HTTP connection timeout.
I don't know why the request headers are absent. Can you provide me some more details about your code? Maybe a code snippet of the client generation? Are you using the httplug handler or the basic soap handler and did you already tried using the other one? What PHP version are you on? ...

Thanks for reporting!

**Comment by dodedodo33** [comment]
@veewee Thanks for your reply!
I had not yet tried using a different http handler, totally forgot about that being an option. After your reply I tried using the httplug handler and I got the same result but a more helpful error message.

It turns out the connection to the host fails (curl error 7), since this service is on a non standard port it is most likely blocked by a firewall. So we'll have to sort this out with the hosting provider.

As to why the debugLastSoapRequest call doesn't show any headers I wouldn't know, maybe because the connection could not be established and the headers were never sent?

Thanks again for your help!
